src/NTestController/NTestController.cs

```diff
@@ -56,16 +58,37 @@ public static void Main(string[] args)
                     // Load plugins.
                     Dictionary<PluginType, IPlugin> plugins = GetPlugins(_options.ConfigFile);
 
+                    var xmlDoc = XmlUtils.LoadXmlDocument(_options.ConfigFile);
+                    var defaultsNode = GetDefaultsXmlNode(xmlDoc);
+                    var platforms = GetPlatforms(xmlDoc, defaultsNode);
+
                     // Execute Reader plugin.
-                    plugins[PluginType.TestReader].Execute();
+                    var readerPlugin = plugins[PluginType.TestReader] as IReaderPlugin;
+                    readerPlugin.TestInputFile = _options.TestFile;
+                    readerPlugin.Execute();
+
+                    // Add all tests to queue.
+                    var testQueue = new TestQueue();
+
+                    foreach (var test in readerPlugin.Tests)
+                    {
+                        testQueue.EnqueueTestToRun(test);

```

Comment:
Maybe the TestQueue should be passed to the Reader plugin (through a property) and then the Execute() function can do this part.